api/python/tests/test_indigo.py

```diff
@@ -117,72 +96,117 @@ def test_check_complex_ion(self) -> None:
-            "{} contains sodium and chloride ions.".format(m1.smiles()),
+            m1.checkSalt(), f"{m1.smiles()} contains sodium and chloride ions."
         )
         self.assertTrue(
             m2.checkSalt(),
-            "{} contains potassium and sulfate ions.".format(m2.smiles()),
+            f"{m2.smiles()} contains potassium and sulfate ions.",
         )
 
     def test_check_no_ions(self) -> None:
         m1 = self.indigo.loadMolecule("c1ccccc1")
         m2 = self.indigo.loadMolecule("C1=CC=C2C=CC=CC2=C1")
         self.assertFalse(
-            m1.checkSalt(), "{} doesn`t contain any salts.".format(m1.smiles())
+            m1.checkSalt(), f"{m1.smiles()} doesn`t contain any salts."
         )
         self.assertFalse(
-            m2.checkSalt(), "{} doesn`t contain any salts.".format(m2.smiles())
+            m2.checkSalt(), f"{m2.smiles()} doesn`t contain any salts."
         )
 
     def test_check_bonded_metal_atom(self) -> None:
         m1 = self.indigo.loadMolecule("CC[Pb](CC)(CC)CC")
         m2 = self.indigo.loadMolecule("C[Al](C)C")
         self.assertFalse(
-            m1.checkSalt(), "{} doesn`t contain any salts.".format(m1.smiles())
+            m1.checkSalt(), f"{m1.smiles()} doesn`t contain any salts."
         )
         self.assertFalse(
-            m2.checkSalt(), "{} doesn`t contain any salts.".format(m2.smiles())
+            m2.checkSalt(), f"{m2.smiles()} doesn`t contain any salts."
         )
 
     def test_check_bonded_acid_group(self) -> None:
         m1 = self.indigo.loadMolecule("C1=CC=C(C=C1)[N+](=O)[O-]")
         m2 = self.indigo.loadMolecule("C(C(=O)O)S(=O)(=O)O")
         self.assertFalse(
-            m1.checkSalt(), "{} doesn`t contain any salts.".format(m1.smiles())
+            m1.checkSalt(), f"{m1.smiles()} doesn`t contain any salts."
         )
         self.assertFalse(
-            m2.checkSalt(), "{} doesn`t contain any salts.".format(m2.smiles())
+            m2.checkSalt(), f"{m2.smiles()} doesn`t contain any salts."
+        )
+
+    def test_salt_stripping(self) -> None:
+        # No disconnected components
+        m1 = self.indigo.loadMolecule("CCCCCCCCCCCCCCCC[N+]1C=CC=CC=1")
+
+        # Single counterion
+        m2 = self.indigo.loadMolecule("CCCCCCCCCCCCCCCC[N+]1C=CC=CC=1.[Cl-]")
+
+        # Many inorganic ions and water molecules
+        m3 = self.indigo.loadMolecule(
+            "CCCCCCCCCCCCCCCC[N+]1C=CC=CC=1.O.O.O.O.O.O.O.O.O.O.[Cl-].[Cl-]"
+        )
+
+        # Two organic molecules with single counterion
+        m4 = self.indigo.loadMolecule(
+            "CCCCCCCCCCCCCCCC[N+]1C=CC=CC=1."
+            "CCCCCCCCCCCCCCCC[N+]1C=CC=CC=1."
+            "[O-]S(=O)(=O)[O-]"
         )
+
+        # Complex organic salt
+        m5 = self.indigo.loadMolecule(
+            "C(C(C(C(C(C(=O)O)O)O)O)O)O.C(C(C(C(C(C(=O)O)O)O)O)O)O.[Fe]"
+        )
+
+        # No organic components
+        m6 = self.indigo.loadMolecule("[NH4+].[O-]P(=O)([O-])[O-].[Fe+2]")
+
```

Comment:
If these `m1`... `m6` are independent, then it's better to make one (independent) test per molecule.
This refers to other tests as well, but they did exist before.

Reply:
All tests were fixed, thank you!